app/controllers/account/admin/admin_controller.rb

```diff
@@ -1,2 +1,13 @@
 class Account::Admin::AdminController < Account::AccountController
+  def start_impersonate
+    user = User.where("id = ? AND role != ?", params[:user_id], 0).take
+    impersonate_user(user) if user.present?
+
```

Comment:
add error message "You can't impersonate this user" if there is no user
don't redirect on error